core/src/main/java/org/elasticsearch/index/query/SpanMultiTermQueryBuilder.java

```diff
@@ -42,6 +60,42 @@ protected void doXContent(XContentBuilder builder, Params params)
     }
 
     @Override
+    protected Query doToQuery(QueryParseContext parseContext) throws IOException {
+        Query subQuery = multiTermQueryBuilder.toQuery(parseContext);
+        if (subQuery instanceof DateFieldMapper.DateFieldType.LateParsingQuery) {
+            subQuery = ((DateFieldMapper.DateFieldType.LateParsingQuery) subQuery).rewrite(null);
```

Comment:
not sure what we should do in master, seems like things are broken either way. This is fine in our branch cause that's how we move forward for now, till we have found a better solution.